packages/mask/src/components/shared/SelectProfileUI/SelectProfileUI.tsx

```diff
@@ -105,7 +116,29 @@ export function SelectProfileUI(props: SelectProfileUIProps) {
         [selected],
     )
 
-    const isEmpty = listBeforeSearch.length > 0 && listAfterSearch.length === 0 && search
+    const results = useMemo(() => {
+        if (!keyword) return items
+
+        return Fuse.create(items, {
+            keys: [
+                'identifier.userId',
+                'nickname',
+                'walletAddress',
+                'linkedPersona.rawPublicKey',
+                'linkedPersona.publicKeyAsHex',
+                'linkedTwitterNames',
+            ],
+            isCaseSensitive: false,
+            ignoreLocation: true,
+            threshold: 0,
+        })
+            .search(keyword)
+            .map((item) => item.item)
+            .filter((x) => !frozenPubkeyList.includes(x.linkedPersona?.publicKeyAsHex!))
+    }, [keyword, items, frozenPubkeyList])
```

Comment:
Do not recreate a Fuse each time the keyword changes.

```ts
const fuse = useMemo(() => {
return Fuse.create(items, ...)
}, [items])

const results = useMemo(() => {
  return fuse.search(keyword)
}, [fuse, keyword]

```

By the way, `frozenPubkeyList` should be memorized above

```diff
-    const frozenPubkeyList = compact(frozenSelected.map((x) => x.linkedPersona?.publicKeyAsHex))
+    const frozenPubkeyList = useMemo(() => compact(frozenSelected.map((x) => x.linkedPersona?.publicKeyAsHex)), [frozenSelected])
```